src/main/java/com/onarandombox/MultiverseCore/commands/GameruleCommand.java

```diff
@@ -64,17 +65,38 @@ public void runCommand(CommandSender sender, List<String> args) {
             world = Bukkit.getWorld(args.get(2));
             if (world == null) {
                 sender.sendMessage(ChatColor.RED + "Failure!" + ChatColor.WHITE + " World " + ChatColor.AQUA + args.get(2)
-                    + ChatColor.WHITE + " does not exist.");
+                        + ChatColor.WHITE + " does not exist.");
                 return;
             }
         }
 
-        if (world.setGameRuleValue(gameRule, value)) {
-            sender.sendMessage(ChatColor.GREEN + "Success!" + ChatColor.WHITE + " Gamerule " + ChatColor.AQUA + gameRule
-                    + ChatColor.WHITE + " was set to " + ChatColor.GREEN + value);
+        if (gameRule == null) {
+            sender.sendMessage(ChatColor.RED + "Failure! " + ChatColor.AQUA + args.get(0) + ChatColor.WHITE
+                    + " is not a valid gamerule.");
         } else {
-            sender.sendMessage(ChatColor.RED + "Failure!" + ChatColor.WHITE + " Gamerule " + ChatColor.AQUA + gameRule
-                    + ChatColor.WHITE + " cannot be set to " + ChatColor.RED + value);
+            boolean success = false;
+
+            try {
+                if (gameRule.getType() == Boolean.class) {
+                    if (value.equalsIgnoreCase("true")) success = world.setGameRule(gameRule, true);
+                    else if (value.equalsIgnoreCase("false")) success = world.setGameRule(gameRule, false);
+                } else if (gameRule.getType() == Integer.class) {
+                    success = world.setGameRule(gameRule, Integer.parseInt(value));
+                } else {
+                    sender.sendMessage(ChatColor.RED + "Failure!" + ChatColor.WHITE + " Gamerule " + ChatColor.AQUA + gameRule.getName()
+                            + ChatColor.WHITE + " isn't supported yet, please let us know about it.");
+                    return;
+                }
+            } catch (NumberFormatException ignored) {}
```

Comment:
Can we restrict the scope of this NumberFormatException try/catch to the part that actually needs it? (line 84). Also, we should probably display an error to the user if it happens.

Reply:
Sounds good, will be done. As for why no error is displayed, that's handled right after (see below).

Reply:
Done, and added a more descriptive error message.